Add serverside version info

TsubasaK111 opened the pull request:
@takayukioda  review me!

TsubasaK111 committed:
feat: add serverside version info to references

TsubasaK111 committed:
refactor: minor header markdown change to standard

takayukioda commented on source/docs/en/reference.md:
```diff
@@ -1,14 +1,52 @@
+- C/C++ v4.8
+- C# v4.4.1
+
+##### Frameworks
```
`Tools` might be better; things listed here are Package Tools and Testing Tools.

TsubasaK111 replied:
Yessir!

takayukioda commented on source/docs/en/reference.md:
```diff
@@ -1,14 +1,52 @@
+- prove TAP::Harness v3.26 and Perl v5.18.2
+- rspec v3.4.4
+
+## Sample Challenges List
```
Since we don't show much of these contents, I'm not sure if we should have these list now.
We should list some challenges that are really published.

TsubasaK111 replied:
I agree.

takayukioda commented on source/docs/en/reference.md:
```diff
@@ -20,24 +58,24 @@ Our recommend Browser is Google Chrome.
+##### API(Server) Challenges
+##### HTML(Frontend) Challenges
+##### Markdown Challenges
+##### Mobile Challenges
```
We shouldn't have mobile and game challenges right now.

TsubasaK111 replied:
Same idea as the other sample challenges, yes? I will also have this removed for the time being.

takayukioda commented:
@TsubasaK111 fix me
I left some notes. Basically same thing for Japanese version too.

TsubasaK111 committed:
refactor: reorganize and remove unready content

- rename reference page to reference_user.md

- update mkdocs.yml to link to reference_users.md

- move content for organizers to a separate file called reference_organizers.md

- de-listed content that is not ready for users to viewed by commenting out pages in mkdocs.yml

takayukioda commented:
I've checked his updated code locally. @TsubasaK111 you can merge after you push your codes.
![LGTM](http://lgtm.in/p/2FmiLQvin)

TsubasaK111 committed:
refactor: remove content plans that are no longer valid

TsubasaK111 committed:
chore: prune redundant content

- although get_started.md is not included in the mkdocs.yml file.

- thus this file is not part of the live docs site

- However this file contains some content that might be migrated to the live pages in the future.

- there was some purely redundant text in this file, and this has been pruned.

TsubasaK111 committed:
chore: rebuild static pages
